src/lib/timeseries.js

```diff
@@ -467,24 +467,16 @@ export const computeMonthsAndCO2s = (timeseries, f) => {
   return { months: formatedMonths, CO2s }
 }
 
-export const computeFirstAndLastDay = timeseries => {
-  let firstDay = getStartDate(timeseries[0])
-  let lastDay = getEndDate(timeseries[0])
-
-  timeseries.forEach(timeserie => {
+export const countUniqDays = timeseries => {
+  const timeserieDays = timeseries.map(timeserie => {
     const startDate = getStartDate(timeserie)
     const endDate = getEndDate(timeserie)
-
-    firstDay = startDate < firstDay ? startDate : firstDay
-    lastDay = endDate > lastDay ? endDate : lastDay
+    // differenceInDays returns full day, so returns 0 if endDate and startDate are in the same day
+    // we need to add 1 in this case
+    return 1 + differenceInDays(endDate, startDate)
   })
 
-  return { firstDay, lastDay }
-}
-
-export const countDays = timeseries => {
-  const { firstDay, lastDay } = computeFirstAndLastDay(timeseries)
-  return differenceInDays(lastDay, firstDay)
+  return timeserieDays.reduce((prev, current) => prev + current, 0)
```

Comment:
I'm not sure about that: if you have two different timeseries on the same day, this will count the day two times, right? I think we want to count each day at most once.

Reply:
effectivement, c'est corrigé